spec/models/chunk_spec.rb

```diff
@@ -1,60 +1,106 @@
 require 'rails_helper'
 
 RSpec.describe Chunk do
-  subject { create(:chunk, content: "subject", embedding_content:) }
+  subject { create(:chunk, excerpt:, embedding_content:) }
 
   let(:document) { create(:document) }
-  let(:embedding_content) { nil }
+  let(:excerpt) { "subject" }
+  let(:embedding_content) { "surround subject sound" }
   let(:chunks) { document.chunks.sort }
 
-  before do
-    create(:chunk, document:)
-    create(:chunk, document:)
-    document.chunks << subject
-    create(:chunk, document:)
-    create(:chunk, document:)
+  describe "#create!" do
+    context "without embedding_content" do
+      let(:embedding_content) { nil }
 
-    document.reload
+      it "fails" do
+        expect { subject }.to raise_error(ActiveRecord::RecordInvalid)
+      end
+    end
   end
 
-  describe "#previous" do
-    it "returns the previous chunk" do
-      expect(subject.previous).to contain_exactly(chunks[1])
+  context "when using chunks" do
+    before do
+      create(:chunk, document:)
+      create(:chunk, document:)
+      document.chunks << subject
+      create(:chunk, document:)
+      create(:chunk, document:)
+
+      document.reload
     end
 
-    it "returns multiple chunks" do
-      expect(subject.previous(2)).to eq(chunks[0..1])
+    describe "#previous" do
+      it "returns the previous chunk" do
+        expect(subject.previous).to contain_exactly(chunks[1])
+      end
+
+      it "returns multiple chunks" do
+        expect(subject.previous(2)).to eq(chunks[0..1])
+      end
+
+      it "stops at the beginning of the list" do
+        expect(subject.previous(3)).to eq(chunks[0..1])
+      end
     end
 
-    it "stops at the beginning of the list" do
-      expect(subject.previous(3)).to eq(chunks[0..1])
+    describe "#next" do
+      it "returns the next chunk" do
+        expect(subject.next).to contain_exactly(chunks[-2])
+      end
+
+      it "returns multiple chunks" do
+        expect(subject.next(2)).to eq(chunks[3..4])
+      end
+
+      it "stops at the end of the list" do
+        expect(subject.next(3)).to eq(chunks[3..4])
+      end
     end
-  end
 
-  describe "#next" do
-    it "returns the next chunk" do
-      expect(subject.next).to contain_exactly(chunks[-2])
+    describe "#embedding_content=" do
+      it "fails since property is readonly" do
```

Comment:
This is a good candidate for a `shared_example` now.

Reply:
N/a since these are gone now. 😄